### PR TITLE
dts: arm: stm32g474: Add i2c4

### PR DESCRIPTION
**Pull request by grillchef**
Add DTS configuration for i2c4.

Signed-off-by: MORGER Patrick <patrick.morger@leica-geosystems.com>

### CODE REVIEW

**Comment by grillchef** [comment]
I will merge the two commits into one...

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for this contribution.
Aside from the cleanup of the commit message (and sign-off addition), I have one comment:
Please move i2c4 node definition in stm32g474.dtsi.
With this proposal i2c4 node is made available to stm32g431 targets, which does not match SoC description.

File: dts/arm/st/g4/stm32g474.dtsi [hunk 1]
```diff
@@ -52,5 +52,18 @@
...
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0x40008400 0x400>;
+			clocks = <&rcc STM32_CLOCK_BUS_APB1_2 0x00000002>;
+			interrupts = <82 0>, <83 0>;
+			interrupt-names = "event", "error";
+			status = "disabled";
+			label= "I2C_4";
+		};
+
```
Comment (KwonTae-young) [inline dts/arm/st/g4/stm32g474.dtsi:67]: I want this to be in front of `spi4`.
The reason is that if you refer to `dts/arm/st/g4/stm32g4.dtsi`, i2c is in front of spi in order.

**dts/arm/st/g4/stm32g4.dtsi**
``` c
		i2c3: i2c@40007800 {
			compatible = "st,stm32-i2c-v2";
			clock-frequency = <I2C_BITRATE_STANDARD>;
			#address-cells = <1>;
			#size-cells = <0>;
			reg = <0x40007800 0x400>;
			clocks = <&rcc STM32_CLOCK_BUS_APB1 0x40000000>;
			interrupts = <92 0>, <93 0>;
			interrupt-names = "event", "error";
			status = "disabled";
			label= "I2C_3";
		};

		spi1: spi@40013000 {
			compatible = "st,stm32-spi-fifo", "st,stm32-spi";
			#address-cells = <1>;
			#size-cells = <0>;
			reg = <0x40013000 0x400>;
			interrupts = <35 5>;
			clocks = <&rcc STM32_CLOCK_BUS_APB2 0x00001000>;
			status = "disabled";
			label = "SPI_1";
		};
```

File: drivers/serial/uart_npcx.c [hunk 2]
```diff
@@ -238,7 +264,9 @@ static void uart_npcx_poll_out(const struct device *dev, unsigned char c)
 
 	inst->UTBUF = c;
 }
+#endif /* !CONFIG_UART_INTERRUPT_DRIVEN */
 
+/* UART api functions */
```
Comment (KwonTae-young) [inline drivers/serial/uart_npcx.c:269]: Contains changes not related to `i2c4`.

**Review by KwonTae-young — CHANGES_REQUESTED** [review]
A commit message cleanup is required. And a little suggestion

File: drivers/serial/uart_npcx.c (hunk 2 above)
Comment (grillchef) [reply drivers/serial/uart_npcx.c]: not related changes removed

File: dts/arm/st/g4/stm32g474.dtsi (hunk 1 above)
Comment (grillchef) [reply dts/arm/st/g4/stm32g474.dtsi:56]: i2c4 is now before the spi

**Review by erwango — CHANGES_REQUESTED** [review]
Please clean up the commit message. Otherwise this seems ok

**Review by erwango — APPROVED** [review]

**Review by KwonTae-young — APPROVED** [review]

**Pull request merged**